client/state/current-user/reducer.js

```diff
@@ -60,28 +61,35 @@ export const currencyCode = createReducer( null, {
  * @param  {Object} action Action payload
  * @return {Object}        Updated state
  */
-export const capabilities = createReducer( {}, {
-	[ SITE_RECEIVE ]: ( state, action ) => {
-		if ( ! action.site.capabilities ) {
-			return state;
-		}
+export const capabilities = withSchemaValidation(
+	capabilitiesSchema,
+	( state = {}, action ) => {
```

Comment:
>it's rare that I think it's a good idea to define a function inside of the argument list of another function call.

I think the difference in how I view this is in reading the code, I'd like to consider this to be one individual unit: a reducer (which happens to be enhanced through utilities). Especially in a reducer file, if we were to split this up, I've suddenly lost confidence in being able to identify / differentiate reducers in the file (yes, they may technically still all be reducers).

While you're correct that the function argument couldn't be tested in isolation in its current form, the fact of the matter is that I have no intention of testing it anyways, so pragmatically I'll accept the caveat and happily accept willingness to make minor adjustments to the code if I ever change my mind in the future.